src/js/common/schemaform/SelectiveArrayField.jsx

```diff
@@ -0,0 +1,203 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'lodash/fp';
+import classNames from 'classnames';
+import Scroll from 'react-scroll';
+
+
+import {
+  toIdSchema,
+  getDefaultFormState,
+  deepEquals
+} from 'react-jsonschema-form/lib/utils';
+
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
+
+
+class SelectiveArrayField extends React.Component {
+  // This fills in an empty item in the array if it has minItems set
+  // so that schema validation runs against the fields in the first item
+  // in the array. This shouldn't be necessary, but there's a fix in rjsf
+  // that has not been released yet
+  componentDidMount = () => {
+    const { schema, formData = [], registry } = this.props;
+    if (schema.minItems > 0 && formData.length === 0) {
+      this.props.onChange(Array(schema.minItems).fill(
+        getDefaultFormState(schema.additionalItems, undefined, registry.definitions)
+      ));
+    }
+  }
+
+  shouldComponentUpdate = (nextProps, nextState) => {
+    return !deepEquals(this.props, nextProps) || nextState !== this.state;
+  }
+
+  onItemChange = (indexToChange, value) => {
+    const newItems = _.set(indexToChange, value, this.props.formData || []);
+    this.props.onChange(newItems);
+  }
+
+  getItemSchema = (index, selectedFields) => {
+    const schema = this.props.schema;
+    const itemSchema = schema.items.length > index
```

Comment:
Yeah, I guess if we're adding new items, we'll have to do a lot more than just this...Okay.